src/software/ai/hl/stp/tactic/shoot_goal_tactic_test.cpp

```diff
@@ -13,8 +13,8 @@ TEST(ShootGoalTacticTest, robot_will_shoot_on_open_net)
     Robot robot = Robot(0, Point(0, 0), Vector(2, -1), Angle::zero(),
                         AngularVelocity::zero(), Timestamp::fromSeconds(0));
     world.mutableFriendlyTeam().updateRobots({robot});
-    BallState ballState(Point(ROBOT_MAX_RADIUS_METERS, 0), Vector(0, 0),
-                        Timestamp::fromSeconds(0));
+    BallStateWithTimestamp ballState(Point(ROBOT_MAX_RADIUS_METERS, 0), Vector(0, 0),
+                                     Timestamp::fromSeconds(0));
```

Comment:
Hmm. Might be worth a discussion. That's somewhat tricky, because the `World` and things that access it somewhat rely on each ball state having a timestamp IIRC. Either way, it would likely require non-trivial changes to the `World` class, and so is out-of-scope for this PR IMO.

Reply:
I am more than happy to have a talk about when / where timestamps are useful and how we can deal with that, but agree that's out of scope for this PR.